### PR TITLE
Add Multi-CSV/DF support in CSV and DataFrame Toolkits

### LINKED ISSUE #1958: How to work with multiple csv files in the same agent session ? is there any option to call agent with multiple csv files, so that the model can interact multiple files and answer us.

### LINKED ISSUE #4423: Do Q/A with csv agent and multiple txt files at the same time.
### Issue you'd like to raise.

I want to do Q/A with csv agent and multiple txt files at the same time. But I do not want to use csv loader and txt loader because they did not perform very well when handling cross file scenario. For example, the model needs to find answers from both the csv and txt file and then return the result.

How should I do it? I think I may need to create a custom agent.

### Suggestion:

_No response_

### PR DESCRIPTION
**Pull request by NickL77**
Add Multi-CSV/DF support in CSV and DataFrame Toolkits
* CSV and DataFrame toolkits now accept list of CSVs/DFs
* Add default prompts for many dataframes in `pandas_dataframe` toolkit

Fixes #1958
Potentially fixes #4423

## Testing
* Add single and multi-dataframe integration tests for `pandas_dataframe` toolkit with permutations of `include_df_in_prompt`
* Add single and multi-CSV integration tests for csv toolkit

## Who can review?

Community members can review the PR once tests pass. Tag maintainers/contributors who might be interested: 
@hwchase17 (worked on these toolkits)

### CODE REVIEW

File: langchain/agents/agent_toolkits/csv/base.py
Comment (hwchase17) [inline langchain/agents/agent_toolkits/csv/base.py:11]: can we do more strict typing here than any? Union[...]

**Review by hwchase17 — COMMENTED** [review]
super interesting. can we add an example notebook?

**Comment by NickL77** [comment]
Made the stricter typing fix.

Added an example to the existing notebooks. lmk if there's anything else to change.

**Review by hwchase17 — APPROVED** [review]
might some slight changes just to make it look nicer, but otherwise lgtm!

**Comment by NickL77** [comment]
Is this just waiting for a maintainer to merge? Also if this change makes it to twitter, would appreciate a s/o `nicholasliu77` :smile:

**Comment by dev2049** [comment]
thanks @NickL77!

**Comment by mrugank-wadekar** [comment]
Hi does multi csv or multi df passing through a list does not work with AgentType ZERO SHOT REACT DESCRIPTION?
I've been trying a lot to pass a path to csv files through  a list or passing df objects as mentioned in the documentation, but it constantly results in 'I need to download the data', seems it cannot read it. It can read it when passed as single string or single df object. I am using latest langchain package 0.0.275. Any help would be appreciated

**Pull request merged**
